tests/test_misc.py

```diff
@@ -36,3 +41,60 @@ def test_environment_files(self) -> None:
         # We do the same for the conda dependency, first a sanity check that everything that is in env is also in dev-ev
         diff_conda_check = list(set(conda_dep_env) - set(conda_dep_devenv))
         assert len(diff_conda_check) == 0
+
+    @pytest.mark.parametrize("deprecation_increment", [-1, 0, 1, None])  # type: ignore
+    @pytest.mark.parametrize("details", [None, "It was completely useless!", "dunnowhy"])  # type: ignore
+    def test_deprecate(self, deprecation_increment: int | None, details: str | None) -> None:
+        """
+        Test the deprecation warnings/errors.
+
+        If the removal_version is larger than the current, it should warn.
+        If the removal_version is smaller or equal, it should raise an error.
+
+        :param deprecation_increment: The version number relative to the current version.
+        :param details: An optional explanation for the description.
+        """
+        warnings.simplefilter("error")
+
+        current_version = geoutils.version.version
+
+        # Set the removal version to be the current version plus the increment (e.g. 0.0.5 + 1 -> 0.0.6)
+        removal_version = (
+            current_version[:-1] + str(int(current_version.rsplit(".", 1)[1]) + deprecation_increment)
+            if deprecation_increment is not None
+            else None
+        )
+
+        # Define a function with no use that is marked as deprecated.
+        @geoutils.misc.deprecate(removal_version, details=details)  # type: ignore
+        def useless_func() -> int:
+            return 1
+
+        # If True, a warning is expected. If False, a ValueError is expected.
+        should_warn = removal_version is None or removal_version > current_version
```

Comment:
Comparing the strings directly will not work, e.g. "0.0.10" > "0.0.8" yields False.
You need to use https://packaging.pypa.io/en/latest/version.html.

Reply:
Actually, no correction needed, looks like this is supported by default by Python now:

```python
removal_version
Out[8]: '0.0.8'
current_version
Out[9]: '0.0.7'
removal_version > current_version
Out[10]: True
removal_version < current_version
Out[11]: False
'0.1' > '0.2'
Out[12]: False
'0.1' < '0.2'
Out[13]: True
'0.1' <= '0.1'
Out[14]: True
```

Reply:
No, you haven't tested my counter-example: "0.0.10" > "0.0.8" does not work. 
As stated [here](https://www.geeksforgeeks.org/string-comparison-in-python/), "The relational operators compare the Unicode values of the characters of the strings from the zeroth index till the end of the string. It then returns a boolean value according to the operator used." The unicode value has nothing to do with the integers.
I know that Erik used that comparison in the past, but we had this discussion already and since then I am aware of this issue. I guess what you copy/pasted was an older piece of code that had never been fixed...

Reply:
We can even try with a simpler example: `"10" > "8"` returns False.